src/QueryEditorForm.tsx

```diff
@@ -36,6 +40,17 @@ export function QueryEditorForm(props: Props) {
     },
   };
 
+  /*
+    State used force the component to re-render when fetching
+    editor field options so they show up in the select component
+  */
+  const [needsUpdate, setNeedsUpdate] = useState(false);
+  useEffect(() => {
+    if (needsUpdate) {
+      setNeedsUpdate(false);
+    }
+  }, [needsUpdate]);
```

Comment:
This state is what fixes the issue, allowing us to force re-render the form component after loading in field options